lib/data/repository/default_marvel_repository.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:dio/dio.dart';
+import 'package:marvel/data/data_sources/local_character_data_source.dart';
+import 'package:marvel/data/data_sources/remote_data_source.dart';
+import 'package:marvel/data/models/local_models/local_character.dart';
+import 'package:marvel/domain/entities/character.dart';
+import 'package:marvel/domain/entities/series.dart';
+import 'package:marvel/domain/exceptions/exceptions.dart';
+import 'package:marvel/domain/repository/marvel_repository.dart';
+import 'package:marvel/data/converters/converter.dart';
+
+class DefaultMarvelRepository implements MarvelRepository {
+  final DioDataSource dioDataSource;
+  final HiveDataSource? hiveDataSource;
+  late int offset;
+  late List<Character> characters;
+
+  DefaultMarvelRepository({this.hiveDataSource, required this.dioDataSource});
+
+  @override
+  Future<List<Character>> getCharacters(int offset) async {
```

Comment:
Code duplication detected in this method 🤓

Reply:
refactored